### PR TITLE
Port all data sources and oracle scripts to the new version

### LINKED ISSUE #1355: Port all data sources and oracle scripts to the new version
Look at #1182 for the removed code. This ticket is to revive them back as data sources / oracle scripts. Also add them to genesis.

### PR DESCRIPTION
**Pull request by prin-r**
Fix #1355

### CODE REVIEW

**Review by prin-r — COMMENTED** [review]
Yahoo data source is currently able to work properly.

File: chain/app/genesis.go [hunk 1]
```diff
@@ -132,14 +177,94 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
 			"Crypto price script (Borsh version)",
 			"Oracle script for getting an average crypto price from many sources encoding parameter by borsh.",
 			"./owasm/res/crypto_price_borsh.wasm",
-			`{"Input": "{ \\"kind\\": \\"struct\\", \\"fields\\": [ [\\"symbol\\", \\"string\\"], [\\"multiplier\\", \\"u64\\"] ] }", "Output": "{ \\"kind\\": \\"struct\\", \\"fields\\": [ [\\"px\\", \\"u64\\"] ] }`,
+			`{"Input": "{ \\"kind\\": \\"struct\\", \\"fields\\": [ [\\"symbol\\", \\"string\\"], [\\"multiplier\\", \\"u64\\"] ] }", "Output": "{ \\"kind\\": \\"struct\\", \\"fields\\": [ [\\"px\\", \\"u64\\"] ] }"}`,
```
Comment (sorawit) [inline chain/app/genesis.go]: Why is this changed?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: owasm/chaintests/yahoo_price/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[package]
```
Comment (sorawit) [inline owasm/chaintests/yahoo_price/Cargo.toml:1]: If it does not work. Do not include it.

**Review by sorawit — DISMISSED** [review]
Remove the #fix if this does not fully fix the issue.

File: chain/app/genesis.go (hunk 1 above)
Comment (prin-r) [reply chain/app/genesis.go]: Because JSON.parse on the previous one is error. VM128:1 Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at <anonymous>:1:6

So the newer just add " at the end to make it parsable.

File: owasm/chaintests/yahoo_price/Cargo.toml (hunk 2 above)
Comment (prin-r) [reply owasm/chaintests/yahoo_price/Cargo.toml:1]: working now

File: chain/app/genesis.go [hunk 3]
```diff
@@ -93,15 +93,60 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
 			"./datasources/binance_price.sh",
 		},
 		{
-			"Open weather",
-			"The script that queries current weather",
+			"Open weather map",
+			"The Script that queries weather information from https://api.openweathermap.org",
```
Comment (taobun) [inline chain/app/genesis.go]: script

File: chain/app/genesis.go [hunk 4]
```diff
@@ -93,15 +93,60 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
...
+			"./datasources/bitcoin_count.sh",
+		},
+		{
+			"Bitcoin block hash",
+			"The script that queries block hash of Bitcoin",
+			"./datasources/bitcoin_hash.sh",
+		},
+		{
+			"Coingecko volume script",
+			"The Script that queries crypto volume from Coingecko",
```
Comment (taobun) [inline chain/app/genesis.go]: script

File: chain/app/genesis.go [hunk 5]
```diff
@@ -132,14 +177,84 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
 			"Crypto price script (Borsh version)",
 			"Oracle script for getting an average crypto price from many sources encoding parameter by borsh.",
 			"./owasm/res/crypto_price_borsh.wasm",
-			`{"Input": "{ \\"kind\\": \\"struct\\", \\"fields\\": [ [\\"symbol\\", \\"string\\"], [\\"multiplier\\", \\"u64\\"] ] }", "Output": "{ \\"kind\\": \\"struct\\", \\"fields\\": [ [\\"px\\", \\"u64\\"] ] }`,
+			`{"Input":"{ \"kind\": \"struct\", \"fields\": [ [\"symbol\", \"string\"], [\"multiplier\", \"u64\"] ] }","Output":"{ \"kind\": \"struct\", \"fields\": [ [\"px\", \"u64\"] ] }"}`,
```
Comment (taobun) [inline chain/app/genesis.go]: Escape schema

File: owasm/Cargo.toml [hunk 6]
```diff
@@ -23,4 +23,14 @@ members = [
     "chaintests/get_env",
     "chaintests/gold_price",
     "chaindebugtests/allocate",
+    "chaintests/alphavantage",
```
Comment (taobun) [inline owasm/Cargo.toml:25]: Move all below chaintest/gold_price

File: owasm/chaintests/gold_price/src/lib.rs [hunk 7]
```diff
@@ -14,7 +14,7 @@ struct Output {
 #[no_mangle]
 fn prepare() {
     // Gold price data source
-    oei::request_external_data(5, 1, "".as_bytes());
+    oei::request_external_data(4, 1, "".as_bytes());
```
Comment (taobun) [inline owasm/chaintests/gold_price/src/lib.rs]: Change back to 5

File: owasm/chaintests/qrng/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+#[no_mangle]
+fn execute_impl(_: Input) -> Output {
+    let outputs = (ext::load_input::<String>(1))
+        .iter()
+        .map(|x1| x1.split(",").map(|x2| x2.parse::<u8>().unwrap()).collect::<Vec<_>>())
+        .collect::<Vec<_>>();
+
+    let mut acc = outputs[0].clone();
```
Comment (taobun) [inline owasm/chaintests/qrng/src/lib.rs]: Use fold for this calculation

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/app/genesis.go (hunk 4 above)
Comment (prin-r) [reply chain/app/genesis.go]: Basically all of The Script ...

File: chain/app/genesis.go (hunk 5 above)
Comment (prin-r) [reply chain/app/genesis.go]: I think it works for Golang. Please see https://play.golang.org/p/XDRqdgRg4Nh

File: chain/app/genesis.go [hunk 9]
```diff
@@ -132,16 +177,86 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
 			"Crypto price script (Borsh version)",
 			"Oracle script for getting an average crypto price from many sources encoding parameter by borsh.",
 			"./owasm/res/crypto_price_borsh.wasm",
-			`{"Input": "{ \"kind\": \"struct\", \"fields\": [ [\"symbol\", \"string\"], [\"multiplier\", \"u64\"] ] }", "Output": "{ \"kind\": \"struct\", \"fields\": [ [\"px\", \"u64\"] ] }"}`,
+			`{"Input": "{\"kind\": \"struct\", \"fields\": [ [\"multiplier\", \"u64\"] ] }","Output": "{ \"kind\": \"struct\", \"fields\": [ [\"px\", \"u64\"] ]}"}`,
```
Comment (taobun) [inline chain/app/genesis.go]: Missing symbol or not change schema

File: chain/app/genesis.go [hunk 10]
```diff
@@ -132,16 +177,86 @@ func GetDefaultDataSourcesAndOracleScripts(owner sdk.AccAddress) json.RawMessage
...
-			`{"Input": "{ \"kind\": \"struct\", \"fields\": [ [\"multiplier\", \"u64\"] ] }","Output": "{ \"kind\": \"struct\", \"fields\": [ [\"px\", \"u64\"] ] }"}`,
+			`{"Input": "{\"kind\": \"struct\", \"fields\": [ [\"multiplier\", \"u64\"] ] }","Output": "{ \"kind\": \"struct\", \"fields\": [ [\"px\", \"u64\"] ] }"}`,
 			`https://ipfs.io/ipfs/QmPheBfYjM4fZ6ngSHYrnDgmapZi9r1i4x5hGFUUyZiP5y`,
 		},
+		{
+			"Alphavantage stock price script",
+			"Oracle script for getting an average stock price from Alphavantage",
+			"./owasm/res/alphavantage.wasm",
+			`{"Input":"{ \"kind\": \"struct\", \"fields\": [ [\"symbol\", \"string\"],[\"api_key\", \"string\"], [\"multiplier\", \"u64\"] ] }","Output":"{ \"kind\": \"struct\", \"fields\": [ [\"px\", \"u64\"] ] }"}`,
+			`https://bandprotocol.com`,
```
Comment (taobun) [inline chain/app/genesis.go]: Can you upload code to ipfs? ask @Benzbeeb how to do that

File: owasm/Cargo.toml (hunk 6 above)
Comment (taobun) [reply owasm/Cargo.toml:25]: sorry for my confusing comment, I just want old older and new items come next gold_price and end with `chaindebugtest/allocate`

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by Benzbeeb — APPROVED** [review]
